transmart-rest-api/grails-app/controllers/org/transmartproject/rest/UserQueryController.groovy

```diff
@@ -38,41 +46,53 @@ class UserQueryController {
 
     def save(@RequestParam('api_version') String apiVersion) {
         def requestJson = request.JSON as Map
-        checkForUnsupportedParams(requestJson, ['name', 'patientsQuery', 'observationsQuery', 'bookmarked'])
+        checkForUnsupportedParams(requestJson, ['name', 'patientsQuery', 'observationsQuery', 'bookmarked',
```

Comment:
Why not just specify an object with these fields as parameter of the function?

Reply:
I am not sure, this is the way a request body is checked and parsed in all the other controllers, I didn't want to refactor this now, just extended the existing way. But if you think it would be beneficial to change it, I can do it.

Reply:
There is not that many places where we actually create/edit data in transmart. The [StorageSystemController](https://github.com/thehyve/transmart-core/blob/master/transmart-rest-api/grails-app/controllers/org/transmartproject/rest/StorageSystemController.groovy) is an example, but that uses the generic `RestfulController` class, that may not be the best choice here.
With so many parameters (that are actually properties of the user query object), it makes sense to wrap them either in a command object, or pass them directly as user query objects.

Reply:
@ewelinagr  It looks applicable to the new code. You might want to consider this.